class VasanthException extends Exception{
    public VasanthException(String string){
        super(string);
    }
}



//throw keyword
public class Main {
	public static void main(String[] args) {
		int i=20;
		int j=0;
		int num[]=new int[5];
		String str= null;
		try{
		    j=18/i;
		    if(j==0)
		       throw new VasanthException("we cant print zero");
		    
		      
		 //   System.out.println(num[5]);
		   // System.out.println(str.length());
		}
		catch(VasanthException e){
		  j=18/1;
		  
	      System.out.println("the num should be greaterthan 0");
		}
		catch(ArrayIndexOutOfBoundsException e){
		        System.out.println("the array has no space");
		}
		catch(Exception e){
		        System.out.println("something went wrong");
		}
		System.out.println(j);
		        
		
	}
}
